### PR TITLE
[Chore](build) remove some unused code and remove some wno

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
1. remove some unused code about spinlock
2. remove some wno and fix warning
3. remove varadic macro usage
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -1209,7 +1209,7 @@ void CreateTableTaskPool::_create_tablet_worker_thread_callback() {
         });
         ADOPT_TRACE(trace.get());
         DorisMetrics::instance()->create_tablet_requests_total->increment(1);
-        TRACE("start to create tablet $0", create_tablet_req.tablet_id);
+        TRACE_WITH_ARG("start to create tablet $0", create_tablet_req.tablet_id);
```
Comment (yiguolei) [inline be/src/agent/task_worker_pool.cpp:1212]: 直接把TRACE 这堆宏删了吧， 直接换成 VLOG

File: be/src/util/trace.h [hunk 2]
```diff
@@ -51,17 +51,21 @@ class Trace;
...
 // Example:
 //  TRACE("Acquired timestamp $0", timestamp);
-#define TRACE(format, substitutions...)                                                \
-    do {                                                                               \
-        doris::Trace* _trace = doris::Trace::CurrentTrace();                           \
-        if (_trace) {                                                                  \
-            _trace->SubstituteAndTrace(__FILE__, __LINE__, (format), ##substitutions); \
-        }                                                                              \
-    } while (0)
+#define TRACE(format)                                                 \
```
Comment (yiguolei) [inline be/src/util/trace.h:54]: 直接删了吧，意义不大了

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p1

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**